openwisp_monitoring/device/base/models.py

```diff
@@ -205,6 +205,9 @@ class AbstractDeviceMonitoring(TimeStampedEditableModel):
             app_settings.HEALTH_STATUS_LABELS['critical'],
         ),
     )
+    disabled_checks = models.BooleanField(
+        default=False, help_text='disables all checks for the device',
+    )
```

Comment:
This looks like something which should not be here.
The related issue description does not mention implementing this feature here.
What is the issue that this feature relates to?

Reply:
@nemesisdesign, you mentioned in the issue that ping checks shall also send two notifications. Hence, I added this to disable the checks related to a device when there are connectivity issues. Let me know, if it's not required.

Reply:
the goal of the issue is just to not send the notifications related to the is_working change, nothing else, can you cite the passage you read where you understood you had to do this please? So I will be more careful next time with the wording to avoid misunderstandings.

Reply:
> * users gets several notifications
> 1.is_working changed to False
> 2.failed ping check
> 3.is_working changed to True when the device goes back up
> 4.ping check successful after upgrade
>
>We need to find a way to avoid this redundancy

This is where, I think there was a miscommunication.